### PR TITLE
Various changes I made prior to pushing out M3

### PR DESCRIPTION
**Pull request by eed3si9n**

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -578,10 +593,16 @@ def customCommands: Seq[Setting[_]] = Seq(
     "clean" ::
       "checkCredentials" ::
       "conscript-configs" ::
+      "interfaceProj/compile" :: // Java project needs to compile first
```
Comment (jsuereth) [inline build.sbt]: Ok, this seems pretty broken then.  Not really a huge fan of sbt-doge if it does this.

Comment (eed3si9n) [inline build.sbt]: This may or may not be needed with the latest sbt-doge. I needed to get a build out first then fix doge.

**Comment by eed3si9n** [comment]
The build failures seems transient:

```
downloading https://oss.sonatype.org/content/repositories/releases/org/scala-lang/scala-compiler/2.9.0/scala-compiler-2.9.0.jar ...
:: problems summary ::
:::: WARNINGS
        module not found: org.scala-lang#scala-library;2.9.0
```

`Launcher> testOnly xsbt.boot.ScalaProviderTest` passes locally.

File: build.sbt [hunk 2]
```diff
@@ -568,20 +576,47 @@ def customCommands: Seq[Setting[_]] = Seq(
...
+   * Because of the way ++ (and its improved version wow) is implemented
+   * precompiled compiler briges are handled outside of doge aggregation on root.
+   * `so compile` handles 2.10.x/2.11.x cross building. 
+   */
   commands += Command.command("release-sbt") { state =>
     // TODO - Any sort of validation
     "clean" ::
       "checkCredentials" ::
       "conscript-configs" ::
+      "precompiled-2_8_2/compile" ::
```
Comment (jsuereth) [inline build.sbt]: This is still bad.  Why can't doge handle this?

**Pull request merged**